packages/form-upload/Upload.js

```diff
@@ -75,7 +113,23 @@ const Upload = ({
         } else {
           upload.start();
         }
-        if (rest.onFileUpload) rest.onFileUpload(upload);
+
+        if (
```

Comment:
We might want to keep `if(rest.onFileUpload) rest.onFileUpload(upload)` first in case more users have custom callbacks defined. Not sure which will be the more favored use case but we can come back to it

Reply:
I do have it here underneath in an `else if` or are you thinking of making it completely separate from the condition?